File: 1975 Maximum Matrix Sum/maxMatrixSum.py
from typing import List

class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        if not matrix or not matrix[0]:
            return 0
        m = abs(matrix[0][0])
        res = 0
        neg_singular = 0
        for row in matrix:
            for num in row:
                res += abs(num)
                if num < 0:
                    neg_singular = 1 - neg_singular
                m = min(m, abs(num))
        return res - neg_singular * 2 * m

sol = Solution()
print(sol.maxMatrixSum([[1,2,3],[-1,-2,-3],[1,2,3]]))

File: problems/1975 Maximum Matrix Sum/max_matrix_sum.rs
impl Solution {
    pub fn max_matrix_sum(matrix: Vec<Vec<i32>>) -> i64 {
        let mut m = i64::MAX;
        let mut res: i64 = 0;
        let mut neg_singular: i64 = 0;
        for row in matrix {
            for num in row {
                let na = num.abs() as i64;
                res += na;
                if na < m {
                    m = na;
                }
                if num < 0 {
                    neg_singular = 1 - neg_singular;
                }
            }
        }
        res - neg_singular * 2 * m
    }
}